### ISSUE TITLE
Cache OCI packages for redeployment

### ISSUE THREAD

**Issue by Racer159** [zarf-dev/zarf#2033]
### Is your feature request related to a problem? Please describe.
When deploying OCI references over an over again (common in dev workflows) you currently need to redownload the package over and over again even if the shasums have not changed.

### Describe the solution you'd like

- **Given** I have an OCI package
- **When** I run `zarf package deploy oci://reference:tag`
- **Then** Zarf will download the layers to cache and unpack them from there instead of tmp.
- **And When** I rerun  `zarf package deploy oci://reference:tag`
- **Then** Zarf will deploy from the cache instead of redownloading (assuming the remote shas for the tag have not changed)

### Describe alternatives you've considered
We could tell people to pull then deploy (and this is a valid work around for now) but it is not quite the experience many people would expect.

### Additional context
We need to consider disk management and may need to provide better ways to look into and prune `zarf` cache as a part of this issue.

**Comment by Noxsios** [comment]
@Racer159 Should there be a package level `--cache-policy` | `--oci-cache-policy`?

**Comment by bburky** [comment]
Misc notes after talking to @Noxsios:

> Should there be a package level --cache-policy | --oci-cache-policy?
I think I would mimic docker/podman/kubernetes's behavior. If you do two back to back `zarf package deploy oci://foo` commands, I would expect this to always use the same image, even if the image has changed in the remote registry. I'd consider maybe mirroring the `docker/podman run --pull=...` syntax:
https://docs.podman.io/en/latest/markdown/podman-run.1.html#pull-policy
https://docs.docker.com/engine/reference/commandline/run/#pull

Docker/podman have a default of `missing` and only pull the image if the tag is not present locally. The [default `imagePullPolicy:` setting of `IfNotPresent`](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy
) is similar. These match what I describe above.

A `--pull=always` may still reuse layers if the digest didn't change or if the layers are unchanged between packages.


There's also the question of when you cache. I think I'd personally say any `oci://` command should probably populate the cache unconditionally (and operating on a zarf tarball would not). This might start causing some increased disk usage for users doing `zarf package deploy oci://...` though, but I'd argue this is the behavior users would expect. Perhaps you may want 

Perhaps you want to grow a `zarf package pull --cache oci://..` command to explicitly populate the cache without writing to a zarf tarball.

**Comment by AustinAbro321** [comment]
Yeah disk usage is the obvious worry here, but at a minimum it makes sense to cache image layers especially since Zarf is most likely going to move to ORAS for image pulling and pushing - #3434. Once that happens it'll be simpler to keep Zarf artifacts in the existing Zarf cache in the same oci-format layout directory.
